### DISCUSSION TITLE
Using deprecated class in serializer

### DISCUSSION THREAD

**Discussion by winfredekpe** [Lambdua/openai4j#82] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/f6fac454-d570-4dc8-8c7c-af59f96908e9)

the ImageContent class is still used in the serializer even though it is marked as deprecated.  since the MultiMediaContent is the new class why not update  the serializer to use it. I had an error about casting when I tried to attach an image  using the MultiMediaContent and it took a while for me to know realise the serializer was still using the ImageContent not the MultiMediaContent. Or am I doing something wrong ?

**Comment by Lambdua** [comment, marked as answer]
`v0.22.8` version should have fixed this bug, you can try updating the version to see, of course, to maintain forward compatibility, I also retained the serialization logic of `ImageContent`
